staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -68,8 +69,8 @@ type HTTPClient interface {
 // ResponseWrapper is an interface for getting a response.
 // The response may be either accessed as a raw data (the whole output is put into memory) or as a stream.
 type ResponseWrapper interface {
-	DoRaw(context.Context) ([]byte, error)
-	Stream(context.Context) (io.ReadCloser, error)
+	DoRaw(...context.Context) ([]byte, error)
```

Comment:
Let's keep the interface as it is (API break!). Passing nil for "no context" seems fine to me.

Reply:
My idea is use a variable argument list for backward compatibility. In this case, the function DoRaw() can have a context paramete `DoRaw(ctx)` or no paramete `DoRaw()`. We can delete the context.Context paramete in the future.
But this is for the user to use, the modification behavior is not controlled. I'll modify it.

Reply:
Done, thanks.

Reply:
Changing an interface can also break code which implements it. This code used to work:
```
var rest.Request = &myRequest{}

type myRequest struct {
...
}

func (r *myRequest) DoRaw(context.Context) ([]byte, error) { ... }
```

With the change to `DoRaw(...context.Context) ([]byte, error)` it no longer does. Extending an interface has the same problem.

This code also breaks:
```
var doRaw func(context.Context) ([]byte, error)

doRaw = r.DoRaw
```

It's not clear whether any of this is actually going to be a problem. But it might, so let's avoid it whenever possible.

Reply:
I have reverted the interface definition. :-)